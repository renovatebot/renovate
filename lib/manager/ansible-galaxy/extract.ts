import * as datasourceGalaxy from '../../datasource/galaxy';
import * as datasourceGitTags from '../../datasource/git-tags';
import { logger } from '../../logger';
import { SkipReason } from '../../types';
import type { PackageDependency, PackageFile } from '../types';

const galaxyUrlRegex = /((git\+)?((git|ssh|http(s)?):\/\/)?(.*@)?([\w.-]+)((:\d+)?\/|:))([\w./-]+)(\.git)?/;
const galaxyRoleRegex = /[\w-]+\.[\w-]+/;

function interpretLine(
  lineMatch: RegExpMatchArray,
  lineNumber: number,
  dependency: PackageDependency
): PackageDependency {
  const localDependency: PackageDependency = dependency;
  const key = lineMatch[2];
  const value = lineMatch[3].replace(/["']/g, '');
  switch (key) {
    case 'name': {
      localDependency.managerData.name = value;
      break;
    }
    case 'version': {
      localDependency.managerData.version = value;
      localDependency.currentValue = value;
      localDependency.managerData.lineNumber = lineNumber;
      break;
    }
    case 'scm': {
      localDependency.managerData.scm = value;
      break;
    }
    case 'src': {
      localDependency.managerData.src = value;
      break;
    }
    default: {
      return null;
    }
  }
  return localDependency;
}

function finalize(dependency: PackageDependency): boolean {
  const dep = dependency;
  if (dependency.managerData.version === null) {
    dep.skipReason = SkipReason.NoVersion;
    return false;
  }

  const source: string = dep.managerData.src;
  const sourceMatch: RegExpMatchArray = galaxyUrlRegex.exec(source);
  if (sourceMatch) {
    dep.datasource = datasourceGitTags.id;
    // remove ending `.git` from URLs
    dep.depName = sourceMatch[10].replace(/\.git$/, '');
    // remove leading `git+` from URLs like `git+https://...`
    dep.lookupName = source.replace(/git\+/, '');
  } else if (galaxyRoleRegex.exec(source)) {
    dep.datasource = datasourceGalaxy.id;
    dep.depName = dep.managerData.src;
    dep.lookupName = dep.managerData.src;
  } else if (galaxyRoleRegex.exec(dep.managerData.name)) {
    dep.datasource = datasourceGalaxy.id;
    dep.depName = dep.managerData.name;
    dep.lookupName = dep.managerData.name;
  } else {
    dep.skipReason = SkipReason.NoSourceMatch;
    return false;
  }
  if (dep.managerData.name !== null) {
    dep.depName = dep.managerData.name;
  }

  return true;
}

export default function extractPackageFile(
  content: string
): PackageFile | null {
  logger.trace('ansible-galaxy.extractPackageFile()');
  const newBlockRegEx = new RegExp(/^\s*-\s*((\w+):\s*(.*))$/);
  const blockLineRegEx = new RegExp(/^\s*((\w+):\s*(.*))$/);
  const deps: PackageDependency[] = [];
  const lines = content.split('\n');
  try {
    for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
      let lineMatch = newBlockRegEx.exec(lines[lineNumber]);
      if (lineMatch) {
        const dep: PackageDependency = {
          managerData: {
            name: null,
            version: null,
            scm: null,
            src: null,
          },
        };
        do {
          const localdep = interpretLine(lineMatch, lineNumber, dep);
          if (localdep == null) {
            break;
          }
          const line = lines[lineNumber + 1];

          if (!line) {
            break;
          }
          lineMatch = blockLineRegEx.exec(line);
          if (lineMatch) {
            lineNumber += 1;
          }
        } while (lineMatch);
        if (finalize(dep)) {
          delete dep.managerData;
          deps.push(dep);
        }
      }
    }

    if (!deps.length) {
      return null;
    }
    return { deps };
  } catch (err) /* istanbul ignore next */ {
    logger.debug({ err }, 'Error extracting ansible-galaxy deps');
    return null;
  }
}
