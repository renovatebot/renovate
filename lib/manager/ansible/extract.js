const { getDep } = require('../dockerfile/extract');

module.exports = {
  extractPackageFile,
};

function extractPackageFile(content) {
  logger.debug('ansible.extractPackageFile()');
  let deps = [];
  let lineNumber = 0;
  for (const line of content.split('\n')) {
    const match = line.match(/^\s*image:\s*'?"?([^\s'"]+)'?"?\s*$/);
    if (match) {
      const currentFrom = match[1];
      const dep = getDep(currentFrom);
      logger.debug(
        {
          depName: dep.depName,
          currentValue: dep.currentValue,
          currentDigest: dep.currentDigest,
        },
        'Docker image inside ansible'
      );
      dep.lineNumber = lineNumber;
      dep.versionScheme = 'docker';
      deps.push(dep);
    }
    lineNumber += 1;
  }
  deps = deps.filter(
    dep => !(dep.currentValue && dep.currentValue.includes('${'))
  );
  if (!deps.length) {
    return null;
  }
  return { deps };
}
