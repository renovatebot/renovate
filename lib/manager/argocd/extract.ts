import { loadAll } from 'js-yaml';
import * as gitTags from '../../datasource/git-tags';
import { HelmDatasource } from '../../datasource/helm';
import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
import type { ApplicationDefinition } from './types';
import { fileTestRegex } from './util';

function createDependency(
  definition: ApplicationDefinition
): PackageDependency {
  const source = definition.spec?.source;

  if (source == null) {
    return null;
  }

  // a chart variable is defined this is helm declaration
  if (source.chart) {
    return {
      depName: source.chart,
      registryUrls: [source.repoURL],
      currentValue: source.targetRevision,
      datasource: HelmDatasource.id,
    };
  }
  return {
    depName: source.repoURL,
    currentValue: source.targetRevision,
    datasource: gitTags.id,
  };
}

export function extractPackageFile(
  content: string,
  fileName: string,
  config?: ExtractConfig
): PackageFile | null {
  // check for argo reference. API version for the kind attribute is used
  if (fileTestRegex.test(content) === false) {
    return null;
  }

  const definitions: ApplicationDefinition[] = loadAll(content);

  const deps = definitions
    .map((definition) => createDependency(definition))
    .filter(Boolean);

  return deps.length ? { deps } : null;
}
