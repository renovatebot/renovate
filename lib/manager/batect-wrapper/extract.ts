import { id as githubReleaseDatasource } from '../../datasource/github-releases';
import { logger } from '../../logger';
import { id as semverVersioning } from '../../versioning/semver';
import type { PackageDependency, PackageFile } from '../types';

const VERSION_REGEX = /^\s+VERSION="(.*)"$/m;

export function extractPackageFile(fileContent: string): PackageFile | null {
  logger.trace('batect.extractPackageFile()');
  const match = VERSION_REGEX.exec(fileContent);

  if (match === null) {
    return null;
  }

  const dependency: PackageDependency = {
    depName: 'batect/batect',
    commitMessageTopic: 'Batect',
    currentValue: match[1],
    datasource: githubReleaseDatasource,
    versioning: semverVersioning,
  };

  logger.trace(dependency, 'Found Batect wrapper version');

  return { deps: [dependency] };
}
