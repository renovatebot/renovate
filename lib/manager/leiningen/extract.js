const { DEFAULT_MAVEN_REPO } = require('../maven/extract');

const DEFAULT_CLOJARS_REPO = 'https://clojars.org/repo/';

function trimAtKey(str, kwName) {
  const regex = new RegExp(`:${kwName}(?=\\s)`);
  const keyOffset = str.search(regex);
  if (keyOffset < 0) return null;
  const withSpaces = str.slice(keyOffset + kwName.length + 1);
  const valueOffset = withSpaces.search(/[^\s]/);
  if (valueOffset < 0) return null;
  return withSpaces.slice(valueOffset);
}

function expandDepName(name) {
  return name.indexOf('/') === -1 ? `${name}:${name}` : name.replace('/', ':');
}

function extractFromVectors(str, offset = 0, ctx = {}) {
  if (str.indexOf('[') !== 0) return [];
  let balance = 0;
  const result = [];
  let idx = 0;
  let vecPos = 0;
  let artifactId = '';
  let version = '';
  let fileReplacePosition = null;

  const isSpace = ch => ch && /[\s,]/.test(ch);

  const cleanStrLiteral = s => s.replace(/^"/, '').replace(/"$/, '');

  const yieldDep = () => {
    if (artifactId && version && fileReplacePosition) {
      result.push({
        ...ctx,
        datasource: 'maven',
        depName: expandDepName(cleanStrLiteral(artifactId)),
        currentValue: cleanStrLiteral(version),
        fileReplacePosition,
      });
    }
    artifactId = '';
    version = '';
  };

  let prevChar = null;
  while (idx < str.length) {
    const char = str.charAt(idx);
    if (char === '[') {
      balance += 1;
      if (balance === 2) {
        vecPos = 0;
      }
    } else if (char === ']') {
      balance -= 1;
      if (balance === 1) {
        yieldDep();
      } else if (balance === 0) {
        break;
      }
    } else if (balance === 2) {
      if (isSpace(char)) {
        if (!isSpace(prevChar)) {
          vecPos += 1;
        }
      } else if (vecPos === 0) {
        artifactId += char;
      } else if (vecPos === 1) {
        if (isSpace(prevChar)) {
          fileReplacePosition = offset + idx + 1;
        }
        version += char;
      }
    }
    prevChar = char;
    idx += 1;
  }
  return result;
}

function extractLeinRepos(content) {
  const result = [DEFAULT_CLOJARS_REPO, DEFAULT_MAVEN_REPO];

  const repoContent = trimAtKey(
    content.replace(/;;.*(?=[\r\n])/g, ''), // get rid of comments
    'repositories'
  );

  if (repoContent) {
    let balance = 0;
    let endIdx = 0;
    for (let idx = 0; idx < repoContent.length; idx += 1) {
      const char = repoContent.charAt(idx);
      if (char === '[') {
        balance += 1;
      } else if (char === ']') {
        balance -= 1;
        if (balance <= 0) {
          endIdx = idx;
          break;
        }
      }
    }
    const repoSectionContent = repoContent.slice(0, endIdx);
    const matches = repoSectionContent.match(/"https?:\/\/[^"]*"/g) || [];
    const urls = matches.map(x => x.replace(/^"/, '').replace(/"$/, ''));
    urls.forEach(url => result.push(url));
  }

  return result;
}

function extractPackageFile(content) {
  const collect = (key, ctx) => {
    let result = [];
    let restContent = trimAtKey(content, key);
    while (restContent) {
      const offset = content.length - restContent.length;
      result = [...result, ...extractFromVectors(restContent, offset, ctx)];
      restContent = trimAtKey(restContent, key);
    }
    return result;
  };

  const registryUrls = extractLeinRepos(content);

  const deps = [
    ...collect('dependencies', {
      depType: 'dependencies',
      registryUrls,
    }),
    ...collect('managed-dependencies', {
      depType: 'managed-dependencies',
      registryUrls,
    }),
    ...collect('dev-dependencies', {
      depType: 'managed-dependencies',
      registryUrls,
    }),
    ...collect('plugins', {
      depType: 'plugins',
      registryUrls,
    }),
    ...collect('pom-plugins', {
      depType: 'pom-plugins',
      registryUrls,
    }),
  ];

  return { deps };
}

module.exports = {
  trimAtKey,
  extractFromVectors,
  expandDepName,
  DEFAULT_CLOJARS_REPO,
  extractPackageFile,
};
