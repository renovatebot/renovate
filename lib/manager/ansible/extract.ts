import { logger } from '../../logger';
import * as dockerVersioning from '../../versioning/docker';
import { PackageDependency, PackageFile } from '../common';
import { getDep } from '../dockerfile/extract';

export default function extractPackageFile(
  content: string
): PackageFile | null {
  logger.trace('ansible.extractPackageFile()');
  let deps: PackageDependency[] = [];
  const re = /^\s*image:\s*'?"?([^\s'"]+)'?"?\s*$/;
  for (const line of content.split('\n')) {
    const match = re.exec(line);
    if (match) {
      const currentFrom = match[1];
      const dep = getDep(currentFrom);
      logger.debug(
        {
          depName: dep.depName,
          currentValue: dep.currentValue,
          currentDigest: dep.currentDigest,
        },
        'Docker image inside ansible'
      );
      dep.versioning = dockerVersioning.id;
      deps.push(dep);
    }
  }
  deps = deps.filter((dep) => !dep.currentValue?.includes('${'));
  if (!deps.length) {
    return null;
  }
  return { deps };
}
