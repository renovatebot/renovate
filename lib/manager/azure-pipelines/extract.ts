import { safeLoad } from 'js-yaml';
import * as datasourceGitTags from '../../datasource/git-tags';
import { logger } from '../../logger';
import { PackageDependency, PackageFile } from '../common';
import { getDep } from '../dockerfile/extract';

interface Container {
  image: string;
}

interface Repository {
  type: 'git' | 'github' | 'bitbucket';
  name: string;
  ref: string;
}

interface Resources {
  repositories: Repository[];
  containers: Container[];
}

interface AzurePipelines {
  resources: Resources;
}

export function extractRepository(
  repository: Repository
): PackageDependency | null {
  if (repository.type !== 'github') {
    return null;
  }

  if (!repository.ref?.startsWith('refs/tags/')) {
    return null;
  }

  return {
    autoReplaceStringTemplate: 'refs/tags/{{newValue}}',
    currentValue: repository.ref.replace('refs/tags/', ''),
    datasource: datasourceGitTags.id,
    depName: repository.name,
    depType: 'gitTags',
    lookupName: `https://github.com/${repository.name}.git`,
    replaceString: repository.ref,
  };
}

export function extractContainer(
  container: Container
): PackageDependency | null {
  if (!container.image) {
    return null;
  }

  const dep = getDep(container.image);
  logger.debug(
    {
      depName: dep.depName,
      currentValue: dep.currentValue,
      currentDigest: dep.currentDigest,
    },
    'Azure pipelines docker image'
  );
  dep.depType = 'docker';

  return dep;
}

export function parseAzurePipelines(content: string): AzurePipelines | null {
  let pkg = null;
  try {
    pkg = safeLoad(content);
  } catch (err) /* istanbul ignore next */ {
    logger.error({ err }, 'Error parsing azure-pipelines content');
    return null;
  }

  if (!pkg || !pkg.resources) {
    return null;
  }

  pkg.resources.containers = pkg.resources.containers || [];
  pkg.resources.repositories = pkg.resources.repositories || [];

  return pkg;
}

export function extractPackageFile(content: string): PackageFile | null {
  logger.trace('azurePipelines.extractPackageFile()');
  const deps: PackageDependency[] = [];

  const pkg = parseAzurePipelines(content);
  if (!pkg) {
    return null;
  }

  // grab the repositories tags
  for (const repository of pkg.resources.repositories) {
    const dep = extractRepository(repository);
    if (dep) {
      deps.push(dep);
    }
  }

  // grab the containers tags
  for (const container of pkg.resources.containers) {
    const dep = extractContainer(container);
    if (dep) {
      deps.push(dep);
    }
  }

  if (!deps.length) {
    return null;
  }
  return { deps };
}
