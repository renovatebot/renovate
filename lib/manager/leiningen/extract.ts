import { DEFAULT_MAVEN_REPO } from '../maven/extract';
import { PackageDependency, PackageFile } from '../common';

export const DEFAULT_CLOJARS_REPO = 'https://clojars.org/repo/';

export function trimAtKey(str: string, kwName: string) {
  const regex = new RegExp(`:${kwName}(?=\\s)`);
  const keyOffset = str.search(regex);
  if (keyOffset < 0) return null;
  const withSpaces = str.slice(keyOffset + kwName.length + 1);
  const valueOffset = withSpaces.search(/[^\s]/);
  if (valueOffset < 0) return null;
  return withSpaces.slice(valueOffset);
}

export function expandDepName(name: string) {
  return name.indexOf('/') === -1 ? `${name}:${name}` : name.replace('/', ':');
}

export interface ExtractContext {
  depType?: string;
  registryUrls?: string[];
}

export function extractFromVectors(
  str: string,
  offset = 0,
  ctx: ExtractContext = {}
): PackageDependency[] {
  if (str.indexOf('[') !== 0) return [];
  let balance = 0;
  const result: PackageDependency[] = [];
  let idx = 0;
  let vecPos = 0;
  let artifactId = '';
  let version = '';
  let fileReplacePosition: number = null;

  const isSpace = (ch: string) => ch && /[\s,]/.test(ch);

  const cleanStrLiteral = (s: string) => s.replace(/^"/, '').replace(/"$/, '');

  const yieldDep = () => {
    if (artifactId && version && fileReplacePosition) {
      result.push({
        ...ctx,
        datasource: 'maven',
        depName: expandDepName(cleanStrLiteral(artifactId)),
        currentValue: cleanStrLiteral(version),
        fileReplacePosition,
      });
    }
    artifactId = '';
    version = '';
  };

  let prevChar = null;
  while (idx < str.length) {
    const char = str.charAt(idx);
    if (char === '[') {
      balance += 1;
      if (balance === 2) {
        vecPos = 0;
      }
    } else if (char === ']') {
      balance -= 1;
      if (balance === 1) {
        yieldDep();
      } else if (balance === 0) {
        break;
      }
    } else if (balance === 2) {
      if (isSpace(char)) {
        if (!isSpace(prevChar)) {
          vecPos += 1;
        }
      } else if (vecPos === 0) {
        artifactId += char;
      } else if (vecPos === 1) {
        if (isSpace(prevChar)) {
          fileReplacePosition = offset + idx + 1;
        }
        version += char;
      }
    }
    prevChar = char;
    idx += 1;
  }
  return result;
}

function extractLeinRepos(content: string) {
  const result = [DEFAULT_CLOJARS_REPO, DEFAULT_MAVEN_REPO];

  const repoContent = trimAtKey(
    content.replace(/;;.*(?=[\r\n])/g, ''), // get rid of comments
    'repositories'
  );

  if (repoContent) {
    let balance = 0;
    let endIdx = 0;
    for (let idx = 0; idx < repoContent.length; idx += 1) {
      const char = repoContent.charAt(idx);
      if (char === '[') {
        balance += 1;
      } else if (char === ']') {
        balance -= 1;
        if (balance <= 0) {
          endIdx = idx;
          break;
        }
      }
    }
    const repoSectionContent = repoContent.slice(0, endIdx);
    const matches = repoSectionContent.match(/"https?:\/\/[^"]*"/g) || [];
    const urls = matches.map(x => x.replace(/^"/, '').replace(/"$/, ''));
    urls.forEach(url => result.push(url));
  }

  return result;
}

export function extractPackageFile(content: string): PackageFile {
  const collect = (key: string, ctx: ExtractContext) => {
    let result: PackageDependency[] = [];
    let restContent = trimAtKey(content, key);
    while (restContent) {
      const offset = content.length - restContent.length;
      result = [...result, ...extractFromVectors(restContent, offset, ctx)];
      restContent = trimAtKey(restContent, key);
    }
    return result;
  };

  const registryUrls = extractLeinRepos(content);

  const deps: PackageDependency[] = [
    ...collect('dependencies', {
      depType: 'dependencies',
      registryUrls,
    }),
    ...collect('managed-dependencies', {
      depType: 'managed-dependencies',
      registryUrls,
    }),
    ...collect('dev-dependencies', {
      depType: 'managed-dependencies',
      registryUrls,
    }),
    ...collect('plugins', {
      depType: 'plugins',
      registryUrls,
    }),
    ...collect('pom-plugins', {
      depType: 'pom-plugins',
      registryUrls,
    }),
  ];

  return { deps };
}
