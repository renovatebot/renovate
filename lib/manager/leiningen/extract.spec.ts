/* eslint-disable no-template-curly-in-string */
import { readFileSync } from 'fs';
import { resolve } from 'upath';
import * as datasourceClojure from '../../datasource/clojure';
import { extractFromVectors, extractPackageFile, trimAtKey } from './extract';

const leinProjectClj = readFileSync(
  resolve(__dirname, `./__fixtures__/project.clj`),
  'utf8'
);

describe('manager/clojure/extract', () => {
  it('trimAtKey', () => {
    expect(trimAtKey('foo', 'bar')).toBeNull();
    expect(trimAtKey(':dependencies    ', 'dependencies')).toBeNull();
    expect(trimAtKey(':dependencies \nfoobar', 'dependencies')).toEqual(
      'foobar'
    );
  });
  it('extractFromVectors', () => {
    expect(extractFromVectors('')).toEqual([]);
    expect(extractFromVectors('[]')).toEqual([]);
    expect(extractFromVectors('[[]]')).toEqual([]);
    expect(extractFromVectors('[[foo/bar "1.2.3"]]')).toEqual([
      {
        datasource: datasourceClojure.id,
        depName: 'foo:bar',
        currentValue: '1.2.3',
      },
    ]);
    expect(
      extractFromVectors('[\t[foo/bar "1.2.3"]\n["foo/baz"  "4.5.6"] ]')
    ).toEqual([
      {
        datasource: datasourceClojure.id,
        depName: 'foo:bar',
        currentValue: '1.2.3',
      },
      {
        datasource: datasourceClojure.id,
        depName: 'foo:baz',
        currentValue: '4.5.6',
      },
    ]);
  });
  it('extractPackageFile', () => {
    expect(extractPackageFile(leinProjectClj)).toMatchSnapshot();
  });
});
