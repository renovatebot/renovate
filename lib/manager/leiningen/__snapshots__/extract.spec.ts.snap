// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager/clojure/extract extractPackageFile 1`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": "1.3.0",
      "datasource": "clojure",
      "depName": "org.clojure:clojure",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "1.0",
      "datasource": "clojure",
      "depName": "org.jclouds:jclouds",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "2.3.1",
      "datasource": "clojure",
      "depName": "net.sf.ehcache:ehcache",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "1.2.15",
      "datasource": "clojure",
      "depName": "log4j:log4j",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "3.0.2",
      "datasource": "clojure",
      "depName": "net.3scale:3scale-api",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "2.8.5",
      "datasource": "clojure",
      "depName": "org.lwjgl.lwjgl:lwjgl",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "2.8.5",
      "datasource": "clojure",
      "depName": "org.lwjgl.lwjgl:lwjgl-platform",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "1.4.0",
      "datasource": "clojure",
      "depName": "org.clojure:clojure",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "1.5.0",
      "datasource": "clojure",
      "depName": "org.clojure:clojure",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "0.2.4",
      "datasource": "clojure",
      "depName": "clj-stacktrace:clj-stacktrace",
      "depType": "dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "0.12.0",
      "datasource": "clojure",
      "depName": "clj-time:clj-time",
      "depType": "managed-dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "1.4.6",
      "datasource": "clojure",
      "depName": "me.raynes:fs",
      "depType": "managed-dependencies",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "1.1.1",
      "datasource": "clojure",
      "depName": "lein-pprint:lein-pprint",
      "depType": "plugins",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "0.1.0",
      "datasource": "clojure",
      "depName": "lein-assoc:lein-assoc",
      "depType": "plugins",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "1.1.1",
      "datasource": "clojure",
      "depName": "s3-wagon-private:s3-wagon-private",
      "depType": "plugins",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "0.0.1",
      "datasource": "clojure",
      "depName": "lein-foo:lein-foo",
      "depType": "plugins",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "0.0.1",
      "datasource": "clojure",
      "depName": "lein-bar:lein-bar",
      "depType": "plugins",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "0.7.1",
      "datasource": "clojure",
      "depName": "cider:cider-nrepl",
      "depType": "plugins",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "1.3.13",
      "datasource": "clojure",
      "depName": "com.theoryinpractise:clojure-maven-plugin",
      "depType": "pom-plugins",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "2.1",
      "datasource": "clojure",
      "depName": "org.apache.tomcat.maven:tomcat7-maven-plugin",
      "depType": "pom-plugins",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
    Object {
      "currentValue": "1.9.68",
      "datasource": "clojure",
      "depName": "com.google.appengine:appengine-maven-plugin",
      "depType": "pom-plugins",
      "registryUrls": Array [
        "https://download.java.net/maven/2",
        "https://oss.sonatype.org/content/repositories/releases",
        "https://blueant.com/archiva/snapshots",
        "https://blueant.com/archiva/internal",
      ],
    },
  ],
}
`;
